### PR TITLE
Deleting project reference when project is DNE

### PR DESCRIPTION
**Pull request by sparshypop7**
plz

### CODE REVIEW

File: contract/controller/src/lib.rs [hunk 1]
```diff
@@ -97,7 +97,7 @@ impl Three0 {
                     entries: Vec::new(),
                     num: 0,
                 }
-        }
+        };
```
Comment (techster10173) [inline contract/controller/src/lib.rs:100]: Remove the semicolon

File: src/views/Dash.tsx [hunk 2]
```diff
@@ -38,13 +39,34 @@ export default function Dash() {
     setProjectDetails(details);
   }
 
+  function validityCheck(isValid: boolean) {
```
Comment (techster10173) [inline src/views/Dash.tsx:42]: Make this an async function...remove the many `.then`'s and use `await` instead.

**Review by techster10173 — CHANGES_REQUESTED** [review]
Fix up the suggested comments before PR can be approved

File: src/views/Dash.tsx [hunk 3]
```diff
@@ -38,13 +39,34 @@ export default function Dash() {
     setProjectDetails(details);
   }
 
+  async function validityCheck(isValid: boolean) {
+    if (isValid) {
+      getProjectDetails();
+    } 
+    else {
+      await window.contract.delete_project({
```
Comment (techster10173) [inline src/views/Dash.tsx:47]: Don't use `.catch` wrap it in a try-catch and handle the error in the catch section

**Review by techster10173 — CHANGES_REQUESTED** [review]

File: src/views/Dash.tsx [hunk 4]
```diff
@@ -38,13 +39,35 @@ export default function Dash() {
...
+      catch (e)  {
+        Swal.fire({
+          title: "Error",
+          text: "There was an issue with deleting the project reference to the project that is DNE",
+          icon: "error",
+          confirmButtonText: "Ok",
+        });
+      }
+      navigate("/");
+      Swal.fire({
```
Comment (techster10173) [inline src/views/Dash.tsx:59]: Put the Swal inside of the `try` block. The code here will always be called regardless of if there is an error or not

**Review by techster10173 — CHANGES_REQUESTED** [review]

**Review by techster10173 — APPROVED** [review]
LGTM!

**Review by srawulwar — APPROVED** [review]
need to follow style guide, rest looks good

**Pull request merged**